src/søknad/steg/4-barnasbosted/BarnetsBostedEndre.tsx

```diff
@@ -85,6 +80,12 @@ const BarnetsBostedEndre: React.FC<Props> = ({
     return b !== barn && b.forelder;
   });
 
+  const andreBarnMedForelderUnik = Array.from(new Set(andreBarnMedForelder.map(b => b.forelder?.id)))
+  .map(id => {
+    if (!id) return null;
+    return andreBarnMedForelder.find(b => b.forelder?.id === id);
+  }).filter(Boolean).map(b => b || barn);
```

Comment:
Den map-en på slutten gjør ingenting, men det er nødvendig for å få Typescript til å skjønne at `andreBarnMedForelderUnik` ikke kan inneholde `null` eller `undefeined`. Av en eller annen grunn skjønner ikke Typescript det når man bare har `.filter(Boolean)`.

Reply:
Hva er det vi ønsker å oppnå her? Hva bruker vi egentlig førsteBarnTilHverForelder til? Er det "valg av forelder" til neste barn?

Reply:
` }).filter(Boolean) as IBarn[];`
er nok mer lesbar, men kanskje ikke helt "trygg".
En annen mulighet er å sjekke "type guard" og verifisere at du har en `id` eller en annen prop på `IBarn` i filter-metoden

Reply:
Kunne kanskje blitt lettere å lese om man trakk dette ut i en funksjon med et meningsfylt navn?

Reply:
@charliemidtlyng Jepp, det er for å liste opp knappene der man kan velge "Samme forelder som X", og hvis flere barn har samme forelder, skal man bare vise én knapp.